src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -858,6 +862,23 @@ double ThermalCluster::getMarketBidCost(uint serieIndex, uint hourInTheYear) con
     }
 }
 
+void ThermalCluster::setProductionCost()
+{
+    auto& modulationCost = modulation[thermalModulationCost];
+    if (costgeneration == Data::setManually)
+    {
+        // alias to the marginal cost
+        for (uint h = 0; h != modulation.height; ++h)
+            productionCost[h] = marginalCost * modulationCost[h];
+    }
+    else
+    {
+        const auto& marginalCostPerHour = costsTimeSeries[0].marginalCostTS;
+        for (uint h = 0; h != modulation.height; ++h)
+            productionCost[h] = marginalCostPerHour[h] * modulationCost[h];
+    }
+}
+
```

Comment:
Not a huge fan of this method

`productionCosts` could be a `std::vector` instead of a raw pointer (the `delete[]` is missing I think)

Also production costs are scenarized, so selecting the 1st column is arbitrary.

Reply:
See #1517